src/github.com/matrix-org/dendrite/cmd/dendrite-client-api-server/main.go

```diff
@@ -34,70 +34,45 @@ import (
 )
 
 var (
-	kafkaURIs            = strings.Split(os.Getenv("KAFKA_URIS"), ",")
-	bindAddr             = os.Getenv("BIND_ADDRESS")
-	logDir               = os.Getenv("LOG_DIR")
-	roomserverURL        = os.Getenv("ROOMSERVER_URL")
-	clientAPIOutputTopic = os.Getenv("CLIENTAPI_OUTPUT_TOPIC")
-	serverName           = gomatrixserverlib.ServerName(os.Getenv("SERVER_NAME"))
-	serverKey            = os.Getenv("SERVER_KEY")
-	accountDataSource    = os.Getenv("ACCOUNT_DATABASE")
-	keyDataSource        = os.Getenv("KEY_DATABASE")
+	logDir     = os.Getenv("LOG_DIR")
+	configPath = flag.String("config", "dendrite.yaml", "The path to the config file, For more information see the config file in this repository")
 )
 
 func main() {
 	common.SetupLogging(logDir)
-	if bindAddr == "" {
-		log.Panic("No BIND_ADDRESS environment variable found.")
-	}
-	if len(kafkaURIs) == 0 {
-		// the kafka default is :9092
-		kafkaURIs = []string{"localhost:9092"}
-	}
-	if roomserverURL == "" {
-		log.Panic("No ROOMSERVER_URL environment variable found.")
-	}
-	if clientAPIOutputTopic == "" {
-		log.Panic("No CLIENTAPI_OUTPUT_TOPIC environment variable found. This should match the roomserver input topic.")
-	}
-	if serverName == "" {
-		serverName = "localhost"
-	}
 
-	cfg := config.ClientAPI{
-		ServerName:           serverName,
-		KafkaProducerURIs:    kafkaURIs,
-		ClientAPIOutputTopic: clientAPIOutputTopic,
-		RoomserverURL:        roomserverURL,
-	}
+	flag.Parse()
 
-	var err error
-	cfg.KeyID, cfg.PrivateKey, err = common.ReadKey(serverKey)
+	cfg, err := config.Load(*configPath)
 	if err != nil {
-		log.Panicf("Failed to load private key: %s", err)
+		log.Fatalf("Invalid config file: %s", err)
 	}
 
-	log.Info("Starting clientapi")
+	log.Info("config: ", cfg)
 
-	roomserverProducer, err := producers.NewRoomserverProducer(cfg.KafkaProducerURIs, cfg.ClientAPIOutputTopic)
+	roomserverProducer, err := producers.NewRoomserverProducer(
+		cfg.Kafka.Addresses, string(cfg.Kafka.Topics.InputRoomEvent),
+	)
 	if err != nil {
-		log.Panicf("Failed to setup kafka producers(%q): %s", cfg.KafkaProducerURIs, err)
+		log.Panicf("Failed to setup kafka producers(%q): %s", cfg.Kafka.Addresses, err)
 	}
 
-	federation := gomatrixserverlib.NewFederationClient(cfg.ServerName, cfg.KeyID, cfg.PrivateKey)
+	federation := gomatrixserverlib.NewFederationClient(
+		cfg.Matrix.ServerName, cfg.Matrix.KeyID, cfg.Matrix.PrivateKey,
+	)
 
-	queryAPI := api.NewRoomserverQueryAPIHTTP(cfg.RoomserverURL, nil)
-	accountDB, err := accounts.NewDatabase(accountDataSource, serverName)
+	queryAPI := api.NewRoomserverQueryAPIHTTP("http://"+string(cfg.Listen.RoomServer), nil)
```

Comment:
I've added a method for getting the roomserver URL to the config so that the HTTP hardcoding is only in one place. 5c21bf5